### PR TITLE
MYGO-5802 Update colors palette for WCAG 2

### PR DESCRIPTION
**Pull request by kyryl-bogach**
[MYGO-5802](https://myworkout.atlassian.net/browse/MYGO-5802)
Color palette: [Figma](https://www.figma.com/file/UZvmUc6RClJVg3yyLch2z2/Myworkout-GO?node-id=0%3A1)

These changes applied to the Login/Signup screens (with some minor tweaks) remove all contrast issues.
Webapp would need some tweaks too.

### CODE REVIEW

File: src/sass/variables.scss [hunk 1]
```diff
@@ -49,8 +49,8 @@ $mw-color-background-primary: $MW_COLOR_NEUTRAL_6;
 $mw-color-background-secondary: $MW_COLOR_NEUTRAL_5;
 $mw-color-border-primary: $MW_COLOR_NEUTRAL_4;
 $mw-color-border-secondary: $MW_COLOR_NEUTRAL_3;
-$mw-color-action: $MW_COLOR_BRAND_1;
-$mw-color-error: $MW_COLOR_ACCENT_2;
+$mw-color-action: $MW_COLOR_BRAND_3;
```
Comment (jp-myworkout) [inline src/sass/variables.scss:52]: If this is primarily used for text, then $MW_COLOR_BRAND_5 should be used. $MW_COLOR_BRAND_3 should be used for icons.

**Comment by kyryl-bogach** [comment]
I think this is very hard to review... I am currently using these changes in the login/singup and webapp manually.

File: src/sass/elements/buttons.scss [hunk 2]
```diff
@@ -3,11 +3,11 @@ Provides classes for buttons
  */
 
 .mw-button {
-  @include mw-button($MW_COLOR_BRAND_1, $MW_COLOR_NEUTRAL_6);
+  @include mw-button(#0EBBC9, $MW_COLOR_GRAY_6);
```
Comment (jp-myworkout) [inline src/sass/elements/buttons.scss:6]: Why is the hexadecimal color used here instead of a color variable?

File: src/sass/elements/forms.scss [hunk 3]
```diff
@@ -10,14 +10,14 @@ Provides classes for form elements
...
   line-height: $mw-line-height;
-  color: $mw-color-text-primary;
+  color: $mw-neutral-text;
   height: 44px;
   background: none;
   outline: none;
   box-shadow: none;
   margin: 0;
   &::placeholder {
-    color: $mw-color-inactive;
```
Comment (jp-myworkout) [inline src/sass/elements/forms.scss:20]: Maybe inactive is a better name than placeholder?

File: src/sass/elements/forms.scss [hunk 4]
```diff
@@ -287,25 +287,25 @@ Provides classes for form elements
...
       width: 36px;
     }
     &:checked + label:after {
       left: 32px;
     }
     &:checked + label:before {
-      background-color: $MW_COLOR_BRAND_1;
-      border-color: $MW_COLOR_BRAND_1;
+      background-color: #0EBBC9;
+      border-color: #0EBBC9;
```
Comment (jp-myworkout) [inline src/sass/elements/forms.scss:308]: Why is this hexadecimal color used here?

File: src/sass/elements/typography.scss [hunk 5]
```diff
@@ -33,11 +33,11 @@ Provides classes for typography (typography for HTML elements defined in src/sas
 }
 .mw-quote {
   line-height: $mw-line-height;
-  color: $MW_COLOR_NEUTRAL_2;
+  color: $MW_COLOR_GRAY_2;
```
Comment (jp-myworkout) [inline src/sass/elements/typography.scss:36]: Use mw-neutral-text for all dark gray text.

File: src/sass/mixins.scss [hunk 6]
```diff
@@ -17,9 +17,9 @@ Provides SASS mixins
   $color-text-hover: $text-color;
   $color-text-active: $text-color;
   $color-border-focus: darken($button-color, $active-darken);
-  $color-background-disabled: $MW_COLOR_NEUTRAL_4;
-  $color-border-disabled: $MW_COLOR_NEUTRAL_4;
-  $color-text-disabled: $MW_COLOR_NEUTRAL_6;
+  $color-background-disabled: $MW_COLOR_GRAY_4;
+  $color-border-disabled: $MW_COLOR_GRAY_4;
```
Comment (jp-myworkout) [inline src/sass/mixins.scss:21]: Shouldn't you use mw-disabled here?

File: src/sass/skins/index.scss [hunk 7]
```diff
@@ -5,32 +5,32 @@ Provides all MSL skin classes: partial styles for HTML elements
...
   filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#FF482A', endColorstr='#FFC255',GradientType=0 );
-  color: $mw-color-inverted;
+  color: $mw-neutral-text-inverted;
 }
 .mw-brand-underline {
   &::after {
     content: " ";
     display: block;
     height: 1px;
-    background: linear-gradient(90deg, $MW_COLOR_BRAND_1, $MW_COLOR_BRAND_2);
```
Comment (jp-myworkout) [inline src/sass/skins/index.scss:20]: Is mw-brand-underline used anywhere? In any case, gradients should be defined as variables also, just like solids.

File: src/sass/skins/index.scss [hunk 8]
```diff
@@ -5,32 +5,32 @@ Provides all MSL skin classes: partial styles for HTML elements
...
 }
 .mw-fallback-css .mw-brand-underline::after {
-  background: $MW_COLOR_BRAND_1;
+  background: #0EBBC9;
 }
 .mw-accent-underline {
   @extend .mw-brand-underline;
   &::after {
-    background: linear-gradient(90deg, $MW_COLOR_ACCENT_1, $MW_COLOR_ACCENT_2);
+    background: linear-gradient(90deg, $MW_COLOR_ORANGE_1, $MW_COLOR_RED_2);
```
Comment (jp-myworkout) [inline src/sass/skins/index.scss:29]: Same here, gradients as variables.

**Comment by hansnn** [comment]
I haven't reviewed yet, but I see that the color constants are used in some style definitions. We have agreed earlier that we should always use the variables instead of the constants, so do you think it would make sense to start doing that everywhere here as well?

File: src/docs-assets/styles.scss [hunk 9]
```diff
@@ -111,37 +99,34 @@ header {
     background: $mw-bg-gradient-2-horizontal;
   }
   &.text-primary {
-    background: $mw-color-text-primary;
+    background: $mw-neutral-text;
   }
   &.text-secondary {
-    background: $mw-color-text-secondary;
+    background: $mw-neutral-gfx;
```
[the remark this answers is not in the document]
Comment (kyryl-bogach) [reply src/docs-assets/styles.scss:105]: Notice that this is only used for the html show case of this library: 

image

File: src/docs-assets/styles.scss [hunk 10]
```diff
@@ -111,37 +99,34 @@ header {
...
   }
   &.bg-primary {
-    background: $mw-color-background-primary;
+    background: $mw-background-primary;
```
[the remark this answers is not in the document]
Comment (kyryl-bogach) [reply src/docs-assets/styles.scss:108]: We have:

```
$mw-background-primary: $MW_COLOR_GRAY_6;
$mw-background-secondary: $MW_COLOR_GRAY_5;
$mw-background-inverted: $MW_COLOR_GRAY_2;
```

Are you suggesting `mw-background-plain` instead of `$mw-background-primary`? If that's what you mean, how should the secondary be named?

File: src/docs-assets/styles.scss [hunk 11]
```diff
@@ -111,37 +99,34 @@ header {
...
+    background: $mw-background-secondary;
   }
-  &.border-secondary {
-    background: $mw-color-border-secondary;
+  &.border {
+    background: $mw-border;
   }
   &.action {
-    background: $mw-color-action;
+    background: $mw-action-gfx;
```
[the remark this answers is not in the document]
Comment (kyryl-bogach) [reply src/docs-assets/styles.scss:117]: Adding gfx and action for the docs: 

image

File: src/sass/elements/buttons.scss (hunk 2 above)
Comment (kyryl-bogach) [reply src/sass/elements/buttons.scss:6]: Good question. Is `#0EBBC9` still something we want to have as a variable?

I think it was used as `MW_COLOR_BRAND_1`. Should we add it as a constant or as a variable here?

File: src/sass/elements/forms.scss (hunk 3 above)
Comment (kyryl-bogach) [reply src/sass/elements/forms.scss:20]: It used to be inactive but looking at that `&::placeholder` rule, it makes more sense to call it `placeholder` IMO.

File: src/sass/mixins.scss (hunk 6 above)
Comment (kyryl-bogach) [reply src/sass/mixins.scss:21]: Indeed. What about `$color-text-disabled`? What variable to use?

image

File: src/sass/skins/index.scss (hunk 7 above)
Comment (kyryl-bogach) [reply src/sass/skins/index.scss:20]: It all depends on if we still want to mantain `$MW_COLOR_BRAND_1` and `$MW_COLOR_BRAND_2`

**Review by jp-myworkout — APPROVED** [review]

File: src/sass/variables.scss [hunk 12]
```diff
@@ -3,6 +3,11 @@ Provides SASS variables with Myworkout design details.
 These values can be updated by client apps when appropriate.
  */
 
+$mw-mobile-breakpoint: 980px;
```
Comment (hansnn) [inline src/sass/variables.scss:6]: Not sure if it actually makes sense to have this as a reusable variable? Like, what does the value actually represent and when/where should it be used?

File: src/sass/variables.scss [hunk 13]
```diff
@@ -3,6 +3,11 @@ Provides SASS variables with Myworkout design details.
...
+// Devices with a pixel ratio of 2 or a dppx of 2 (i.e. 192 dpi).
+// 0.5px measures won't work for all devices or browsers like Safari and Firefox.
+$min-2x-pixel-ratio: "(-webkit-min-device-pixel-ratio: 2), (min-resolution: 2dppx)";
```
Comment (hansnn) [inline src/sass/variables.scss:9]: I guess we don't have to introduce this here now since we will remove the 0.5px thing anyway, and I'm pretty sure that this is only used for that

File: src/sass/variables.scss (hunk 12 above)
Comment (kyryl-bogach) [reply src/sass/variables.scss:6]: It's one of the variables that we have repeatead in the API and in the Webapp.

Comment (hansnn) [reply src/sass/variables.scss:6]: I know, but I still have the question "what does the it actually represent and when/where should it be used"?

In the Webapp the variable represents when it should go from "desktop mode" with a left-hand menu to "mobile mode" where it replaces the left-hand menu with a hamburger menu button. 
While in the API it's only used in a very specific use-case, and we use a different breakpoint to decide when we should go from "desktop mode" where the login form is centered to "mobile mode" where the login form is full-width.

In other words: Yes, We have the same variable in both places, but it seems to be used for different things. So does it make sense to have it re-usable then? If it changes in 1 place, should it also change in the other?

Comment (kyryl-bogach) [reply src/sass/variables.scss:6]: > If it changes in 1 place, should it also change in the other?
Probably not 👍🏻. Also, why would we change it? When mobile devices are wider? 🤣 

Removing it 😄

**Review by hansnn — APPROVED** [review]
